### PR TITLE
Yaml PHP extension

### PR DESCRIPTION
**Pull request by EugenMayer**
- using the new _ version
- using the generic template for the configuration
- including configuration of libyaml

### CODE REVIEW

File: manifests/extension/yaml.pp [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Installs a php extension for a specific version of php.
+#
+# Usage:
+#
+#     php::extension::apc { 'apc for 5.4.10':
```
Comment (mattheath) [inline manifests/extension/yaml.pp]: Could this example be updated to match the module?

File: manifests/extension/yaml.pp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#       version   => '3.1.13'
+#       php       => '5.4.10',
+#     }
+#
+class libyaml {
```
Comment (mattheath) [inline manifests/extension/yaml.pp]: Ideally this class would be pulled out into a separate file in `manifests/dependencies` like in boxen/puppet-php#5

File: manifests/extension/yaml.pp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  package { 'libyaml': }
+}
+
+define php::extension::yaml(
+  $version = '1.1.0',
+  $php
+) {
+  require php::config  
+  include libyaml
```
Comment (mattheath) [inline manifests/extension/yaml.pp]: This should probably be a `require libyaml` to ensure it's installed before compiling the `php_extension` below

**Comment by mattheath** [comment]
@EugenMayer thanks for this! I've put a few comments in line, also could you run puppet lint over it to pick up a few errors travis has picked up: https://travis-ci.org/mattheath/puppet-php/builds/9186322

**Comment by mattheath** [comment]
Also could you rebase this to bring it up to date? :smile:

**Comment by EugenMayer** [comment]
Importat stuff first:
Did as you adviced.

Hey mattheath, sorry for all this code-style issues. Iam very new to puppet syntax, but even worse is, that i dont use anything which does the code-formatting. Usually i keep hammering cmd + shit + l when writing new code..but well, that needs a proper editor. Went form textmate to sublime for puppet, but it seems that vim will be the only with code formatting.

**Comment by EugenMayer** [comment]
lets close this, due to the rebase, i needed to create a new branch

**Pull request closed without merging**